### ISSUE TITLE
Android 15 ghost updates

### ISSUE THREAD

**Issue by timkgh** [anod/AppWatcher#107]
Sometimes when I refresh to check for new updates it shows a couple of available updates, then if I refresh again, the updates are either completely gone or replaced by a subset or by other updates. Keep refreshing and the list changes or it shows there are now updates. Going to the Play Store for these updates doesn't show that new versions of the apps are available. This started happening either with the Android 15 upgrade or maybe with some recent AppWatcher app update.

**Comment by timkgh** [comment]
@anod AppWatcher has become unreliable. I get notifications there are 25 app updates but there aren't any. If I refresh the number changes, or it goes down to 0 or then again to seemingly a random number.
Is this an Android 15 issue?

**Comment by anod** [comment]
The behavior of generating device id was changed in Android 15, this id is used to detect what apps supported by the device... I've implemented, but there still some issues

**Comment by mxxcon** [comment]
I just upgraded to Pixel 9 which has Android 15 and it's experiencing the same problem. Every time I refresh it shows me that supposedly a different app has update.

**Comment by timkgh** [comment]
Is it possible that now AppWatcher is also causing a ghost device to show up in the Play Store?
https://play.google.com/library/devices

I see a phone without a carrier that gets a fresh timestamp for "Last Used" and "Registered On" every day.

![95802](https://github.com/user-attachments/assets/82dc9b28-fb4a-45ce-94ce-bd5df569c92b)

**Comment by mxxcon** [comment]
Same here. My Pixel 9 Pro shows up twice, once with my carrier and one "No carrier".

**Comment by timkgh** [comment]
I've uninstalled AppWatcher for now, let's see if the _ghost_ Pixel 9 drops off from the list eventually.
The crazy part is that the actual Pixel 9 does **not** show up at all in the list. But the phone is registered and logged in to my Google account, the Play Store app on the phone itself works fine, I can install apps, etc. It makes it impossible though to send an app from the _web_ Play Store to the phone.

**Comment by anod** [comment]
Interesting, there are some hardcoded values that look like they needs to be initialized correctly

**Comment by mxxcon** [comment]
> The crazy part is that the actual Pixel 9 does **not** show up at all in the list. But the phone is registered and logged in to my Google account, the Play Store app on the phone itself works fine, I can install apps, etc. It makes it impossible though to send an app from the _web_ Play Store to the phone.

I'm also seeing somewhat similar behavior. I also have Galaxy S20FE (Android 13) from which I upgraded to this pixel at the start of December, and even older LG V30 (Android 9).
S20FE is missing from that list but V30 is there. And on both of them AppWatcher works fine. It's only Pixel that has this problem.
![image](https://github.com/user-attachments/assets/1a1b5ff9-c74d-4f1e-a5aa-417e55f8a62a)

I'm not saying AppWatcher caused this. I don't know what's happening there.

**Comment by timkgh** [comment]
@anod I have confirmed that AppWatcher creates lots and lots of similar devices in the Play Store. I did a Google Takeout and I looked at `Google Play Store/Devices.json`. I have 137 similar "no carrier" phones over the past 60 days. They stopped when I uninstalled AppWatcher. Good news is that they seem to drop off after 60 days and the real devices will eventually start showing up in the Play Store.

Meanwhile, I strongly recommend that you somehow mark the app incompatible with Android 15 or have a big warning in the Play Store for people to not install it until fixed.

**Comment by mxxcon** [comment]
Have there been any progress with this bug?

**Comment by mxxcon** [comment]
Any chance this bug could get some attention? It's still happening. I'm now on Android 17.

**Comment by anod** [comment]
Will check, do you see ghost devices in play store?

**Comment by mxxcon** [comment]
<img width="768" height="262" alt="Image" src="https://github.com/user-attachments/assets/fa17e977-002a-408a-8127-9b185b8187c9" />
Yes.

But my bigger pain point are ghost updates. Right now the app shows 4 updates and 2 recently discovered apps. All 4 are already up to date and 2 discovered have been installed for ages.
Every time I refresh it shows different sets of apps.

**Comment by mxxcon** [comment]
I'm not sure if this is fully fixed or it's a different issue. I'm running 1.7.1 and now there are 2 apps always in my list no matter how many times I refresh. Play store shows no updates for these apps.

<img width="768" height="1713" alt="Image" src="https://github.com/user-attachments/assets/c90c7e4f-0a37-4d11-bf06-511f47e08595" />

**Comment by mxxcon** [comment]
I just updated to version 1.7.1, but the problem still doesn't seem to be resolved; in fact, it appears to be somewhat worse now. The app no longer detects updates available in the Play Store for the apps that are on my watch list. 
<img width="960" height="2142" alt="Image" src="https://github.com/user-attachments/assets/3cc129e9-3032-4e78-a8de-757950ed994c" />

**Comment by anod** [comment]
For me it actually detects more updates than google play shows... app watcher refresh doesn't help?

**Comment by mxxcon** [comment]
No, refreshing doesn't make any difference. It still shows non-existent updates and missing real ones.

<img width="960" height="2142" alt="Image" src="https://github.com/user-attachments/assets/0765c8bb-68d5-42e3-8eba-43ebc6bf753a" />
